DMCompiler/DM/Builders/DMExpressionBuilder.cs

```diff
@@ -351,6 +359,8 @@ public enum ScopeMode {
                 return new Args(identifier.Location);
             case "__TYPE__":
                 return new ProcOwnerType(identifier.Location);
+            case "__IMPLIED_TYPE__":
+                return new ConstantPath(identifier.Location, dmObject, inferredPath.GetValueOrDefault());
```

Comment:
Should be an explicit error if `inferredPath` is null
```suggestion
                if (inferredPath == null) {
                    DMCompiler.Emit(WarningCode.BadExpression, identifier.Location,
                        "__IMPLIED_TYPE__ cannot be used here, there is no type being implied");
                    return new Null(identifier.Location);
                }
                
                return new ConstantPath(identifier.Location, dmObject, inferredPath.Value);
```